File: x_a225(love_sorting)/C++/love_sorting.cpp
#include <iostream>
#include <algorithm>

int main(void) {
  int num_of_nums;
  while (std::cin >> num_of_nums) {
    int numbers[num_of_nums];
    for (int i = 0; i < num_of_nums; i++) {
      std::cin >> numbers[i];
    }

    std::sort(numbers, numbers + num_of_nums, [](int former, int latter) {
      if(former % 10 != latter % 10) return former % 10 < latter % 10;
      else return former / 10 > latter / 10;
    });
    
    for (int n: numbers) {
      std::cout << n << " ";
    }
    std::cout << '\n';
  }

  return 0;
}

File: a225(love_sorting)/Python/love_sorting.py
from functools import cmp_to_key

def compare(former: int, latter: int) -> int:
    if former % 10 == latter % 10:
        return latter - former
    return (former % 10) - (latter % 10)

while True:
    try:
        num_of_nums = int(input())
        numbers = list(map(int, input().split()))
        numbers.sort(key = cmp_to_key(compare))
        print(*numbers)
    except: break
